src/cpu/cgroup_v2.rs

```diff
@@ -7,9 +7,29 @@ use std::path::Path;
 const CPU_SYS_V2_NUMBER_OF_FIELDS: usize = 3;
 
 #[cfg(target_os = "linux")]
-pub fn read_and_parse_v2_sys_stat(path: &Path) -> Result<CgroupCpuMeasurement> {
-    let time = precise_time_ns();
+pub fn read_and_parse_v2_sys_stat(
+    path: &Path,
+    cpu_max_path: &Path,
+) -> Result<CgroupCpuMeasurement> {
+    // If the cpu.max file exists, we can use it to calculate the number of CPUs
+    // in the cgroup. It's also required that the first value is not set to "max",
+    // otherwise we can't calculate the number of CPUs.
+    let mut cpu_count = 0;
+    if cpu_max_path.exists() {
+        let reader = file_to_buf_reader(&cpu_max_path)?;
+        for line in reader.lines() {
```

Comment:
Does it make sense to iterate through lines? We only expect exactly one line. Perhaps if there is more or less than one line we should treat it as a parsing error.